tests/components/amberelectric/test_coordinator.py

```diff
@@ -38,82 +40,84 @@ def mock_api_current_price() -> Generator:
+        channels=[
             Channel(identifier="E1", type=ChannelType.GENERAL, tariff="A100"),
-            Channel(identifier="E2", type=ChannelType.CONTROLLED_LOAD, tariff="A180"),
+            Channel(identifier="E2", type=ChannelType.CONTROLLEDLOAD, tariff="A180"),
         ],
-        "Jemena",
-        SiteStatus.ACTIVE,
-        date(2021, 1, 1),
-        None,
+        network="Jemena",
+        status=SiteStatus("active"),
+        activeFrom=date(2021, 1, 1),
+        closedOn=None,
     )
     general_and_feed_in = Site(
-        GENERAL_AND_FEED_IN_SITE_ID,
-        "11111111113",
-        [
+        id=GENERAL_AND_FEED_IN_SITE_ID,
+        nmi="11111111113",
+        channels=[
             Channel(identifier="E1", type=ChannelType.GENERAL, tariff="A100"),
-            Channel(identifier="E2", type=ChannelType.FEED_IN, tariff="A100"),
+            Channel(identifier="E2", type=ChannelType.FEEDIN, tariff="A100"),
         ],
-        "Jemena",
-        SiteStatus.ACTIVE,
-        date(2021, 1, 1),
-        None,
+        network="Jemena",
+        status=SiteStatus("active"),
+        activeFrom=date(2021, 1, 1),
+        closedOn=None,
     )
     instance.get_sites.return_value = [
         general_site,
         general_and_controlled_load,
         general_and_feed_in,
     ]
 
-    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+    with patch("amberelectric.AmberApi", return_value=instance):
         yield instance
 
 
 def test_normalize_descriptor() -> None:
     """Test normalizing descriptors works correctly."""
     assert normalize_descriptor(None) is None
-    assert normalize_descriptor(Descriptor.NEGATIVE) == "negative"
-    assert normalize_descriptor(Descriptor.EXTREMELY_LOW) == "extremely_low"
-    assert normalize_descriptor(Descriptor.VERY_LOW) == "very_low"
-    assert normalize_descriptor(Descriptor.LOW) == "low"
-    assert normalize_descriptor(Descriptor.NEUTRAL) == "neutral"
-    assert normalize_descriptor(Descriptor.HIGH) == "high"
-    assert normalize_descriptor(Descriptor.SPIKE) == "spike"
+    assert normalize_descriptor(PriceDescriptor.NEGATIVE) == "negative"
+    assert normalize_descriptor(PriceDescriptor.EXTREMELYLOW) == "extremely_low"
+    assert normalize_descriptor(PriceDescriptor.VERYLOW) == "very_low"
+    assert normalize_descriptor(PriceDescriptor.LOW) == "low"
+    assert normalize_descriptor(PriceDescriptor.NEUTRAL) == "neutral"
+    assert normalize_descriptor(PriceDescriptor.HIGH) == "high"
+    assert normalize_descriptor(PriceDescriptor.SPIKE) == "spike"
 
 
 async def test_fetch_general_site(hass: HomeAssistant, current_price_api: Mock) -> None:
     """Test fetching a site with only a general channel."""
 
-    current_price_api.get_current_price.return_value = GENERAL_CHANNEL
+    current_price_api.get_current_prices.return_value = GENERAL_CHANNEL
     data_service = AmberUpdateCoordinator(hass, current_price_api, GENERAL_ONLY_SITE_ID)
```

Comment:
Side note: We should not interact with integration details like the coordinator in the tests.

Here we would just set up the integration while patching the library and assert that the expected states are set in the core state machine during different conditions. Use our time test helpers, eg freeze gun, to trigger interval updates.

Please address this in a future PR.

Reply:
Have you got an example handy?

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/e89c007a38f1d4798703b3e4cf9ef3e24d40066d/tests/components/tessie/test_coordinator.py#L30-L42